src/Catalyst.Core.Modules.Ledger/Ledger.cs

```diff
@@ -256,5 +296,117 @@ public void Dispose()
         }
+
+        class ReceiptDeltaTracer : ITxTracer
+        {
+            readonly Delta _delta;
+            readonly long _deltaNumber;
+            readonly IHashProvider _hashProvider;
+            readonly Keccak _deltaHash;
+            readonly List<TransactionReceipt> _txReceipts;
+            int _currentIndex;
+
+            public ReceiptDeltaTracer(Delta delta, Cid deltaHash, long deltaNumber, IHashProvider hashProvider)
+            {
+                _delta = delta;
+                _deltaNumber = deltaNumber;
+                _hashProvider = hashProvider;
+                _deltaHash = new Keccak(deltaHash.Hash.Digest);
+                _txReceipts = new List<TransactionReceipt>(delta.PublicEntries.Count);
+            }
+
+            public IEnumerable<TransactionReceipt> Receipts => _txReceipts;
+
+            public bool IsTracingReceipt => true;
+
+            public void MarkAsSuccess(Address recipient, long gasSpent, byte[] output, LogEntry[] logs)
+            {
+                _txReceipts.Add(BuildReceipt(recipient, gasSpent, StatusCode.Success, logs));
+            }
+
+            public void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error)
+            {
+                _txReceipts.Add(BuildFailedReceipt(recipient, gasSpent));
+            }
+
+            private TransactionReceipt BuildFailedReceipt(Address recipient, long gasSpent)
+            {
+                return BuildReceipt(recipient, gasSpent, StatusCode.Failure, LogEntry.EmptyLogs);
+            }
+
+            private TransactionReceipt BuildReceipt(Address recipient, long spentGas, byte statusCode, LogEntry[] logEntries)
+            {
+                PublicEntry entry = _delta.PublicEntries[_currentIndex];
+
+                TransactionReceipt txReceipt = new TransactionReceipt
+                {
+                    Logs = logEntries,
+                    GasUsedTotal = _delta.GasUsed,
+                    StatusCode = statusCode,
+                    Recipient = entry.IsContractDeployment ? null : recipient,
+                    DeltaHash = _deltaHash,
+                    DeltaNumber = _deltaNumber,
+                    Index = _currentIndex,
+                    GasUsed = spentGas,
+                    Sender = GetAccountAddress(entry.SenderAddress),
+                    ContractAddress = entry.IsContractDeployment ? recipient : null,
+                    DocumentId = entry.GetDocumentId(_hashProvider)
+                };
+
+                _currentIndex += 1;
+
+                return txReceipt;
+            }
+
+            private static Address GetAccountAddress(ByteString publicKeyByteString)
+            {
+                if (publicKeyByteString == null || publicKeyByteString.IsEmpty)
+                {
+                    return null;
+                }
+
+                return publicKeyByteString.ToByteArray().ToKvmAddress();
+            }
+
+            public bool IsTracingActions => false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private property 'IsTracingActions'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)